lib/overview.dart

```diff
@@ -0,0 +1,118 @@
+                              child: Text(
+                                  snapshot.data![index]
+                                      ["name_" + prefs.getString('language')!],
+                                  style: TextStyle(
+                                      fontSize: 30,
+                                      fontWeight: FontWeight.bold)),
+                              alignment: Alignment.centerLeft),
+                        ),
+                        Container(
+                          child: Align(
+                              child: Text(
+                                  categories[snapshot.data![index]["class"]],
+                                  style: TextStyle(
+                                      fontSize: 20, color: Colors.black38)),
+                              alignment: Alignment.centerLeft),
+                        ),
+                        Divider(color: Colors.black, height: 40),
+                        Container(
+                          padding: EdgeInsets.fromLTRB(0, 0, 0, 8.0),
+                          child: Align(
+                              child: Text(
+                                  AppLocalizations.of(context)!.spectrapeaks,
+                                  style: TextStyle(
+                                      fontSize: 20, color: Colors.black)),
+                              alignment: Alignment.centerLeft),
+                        ),
+                        ListView.builder(
+                            shrinkWrap: true,
+                            physics: ClampingScrollPhysics(),
+                            itemCount: snapshot.data![index]["peaks"].length,
+                            itemBuilder: (BuildContext context, int number) {
+                              return Container(
+                                  padding: EdgeInsets.fromLTRB(0, 3.0, 0, 3.0),
+                                  child: Text(
+                                    ("${snapshot.data![index]["peaks"][number]["wavelength"]}" + AppLocalizations.of(context)!.spectradescpart1 + "${snapshot.data![index]["peaks"][number]["intensity"]}" + AppLocalizations.of(context)!.spectradescpart2),
+                                  style: TextStyle(color: Colors.black54, fontSize: 16)
+                                    ,));
+                            }),
+                        Padding(padding: EdgeInsets.only(bottom: 70))
+                      ],
+                    ),
+                  )
+                ],
+              ),
+              floatingActionButtonLocation: FloatingActionButtonLocation.centerFloat,
+              floatingActionButton: OutlinedButton.icon(
+                onPressed: () async {
+                  Uri url = Uri.parse(snapshot.data![index]["wikipedia_page"]);
+                  if (await canLaunchUrl(url)) {
+                    await launchUrl(url, mode: launchMode);
+                  } else {
+                    throw 'Could not launch $url';
+                  }
```

Comment:
I see this pattern very often. Wouldn't it be better for a utility class to contain this?